Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -91,59 +91,58 @@ @interface Branch() <BNCDebugConnectionDelegate, UIGestureRecognizerDelegate, BN
 
 @implementation Branch
 
-static Branch *currInstance;
-
 #pragma mark - Public methods
 
 
 #pragma mark - GetInstance methods
 
 + (Branch *)getInstance {
-    return [Branch getBranchInstance:YES];
+    NSString *branchKey = [BNCPreferenceHelper getBranchKey:YES];
+    if (!branchKey || [branchKey isEqualToString:NO_STRING_VALUE]) {
```

Comment:
Actually, having a check here just to have the appropriate warnings would be good. If they specify get test instance, but they have a live key, that's not gonna work.